### DISCUSSION TITLE
Cpu usage

### DISCUSSION THREAD

**Discussion by Osama-NJ** [ultrafunkamsterdam/undetected-chromedriver#1613] [General]
Hello ! I have noticed that when i am using the undetected driver for couple of hours that there are unusual high usage of cpu that reached more than 50%, for the first 40 min the usage is normal  but after that the issue starts and note that i am using it on headless mode and i viewed it on GUI as well(headless off ) but issue is still there. i am closing the driver after every 5 min to clear the cookies but somehow the driver keeps running in the background even if i closed the session and turned off the whole script, i got the "Continue running background apps when Google Chrome is closed" chrome setting off as well. Tried it on different pcs and have the same issue as well and note that pcs have high end cpu's .

Does anyone have this issue or just me ? 

![usage screenshot](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/76047819/496ced3c-c349-4537-a790-a7f79e774728)

Thanks!

**Comment by riotinri** [comment]
same here

**Comment by ongbanbanhbao** [comment]
Using  chrome_options.add_argument('--disable-gpu')

**Reply by Osama-NJ** [reply]
using "chrome_options.add_argument('--disable-gpu')" will only disable GPU acceleration and improves the performance if you have a low gpu card, but it's irrelevant to my case cuz the issue that i am having it's regards to the cpu usage not gpu usage.

**Reply by ongbanbanhbao** [reply]
The root cause is that your memory is being leaked by the 'close driver' method, and it has not been fixed yet. If you ever notice it, after your script is done and Chrome is closed, Chrome still runs in the background.

I tried adding the option, and it worked for me. Try it and come back to thank me later.

**Reply by Osama-NJ** [reply]
Oh that's interesting!, will try that and see, thanks alot :)

**Comment by Osama-NJ** [comment]
This got solved by using "chrome_options.add_argument('--disable-gpu')", thanks to @ongbanbanhbao for providing this solution much appreciated!!.

btw I had a problem running my scripts in headless mode but i used these options and it worked, for anyone  having this issue:
 ......................................................................................................................................................................................................

            chrome_options.add_argument('--headless')
            chrome_options.add_argument("--nogpu")
            chrome_options.add_argument("--no-sandbox")
            chrome_options.add_argument("--enable-javascript")

            ua = UserAgent()
            userAgent = ua.random
            
            driver.execute_script("Object.defineProperty(navigator, 'driver', {get: () => undefined})")
            driver.execute_cdp_cmd('Network.setUserAgentOverride', {"userAgent": userAgent})
......................................................................................................................................................................................................

there are specific websites that detect headless browsers thus headless not working, so these options will change the user agent for your browser and make it looks like a normal one.
